src/trackedit/itrackeditproject.h

```diff
@@ -24,24 +24,31 @@ class ITrackeditProject
     virtual ~ITrackeditProject() = default;
 
     virtual std::vector<TrackId> trackIdList() const = 0;
-    virtual muse::async::NotifyList<Track> trackList() const = 0;
+    virtual std::vector<Track> trackList() const = 0;
     virtual Clip clip(const ClipKey& key) const = 0;
     virtual muse::async::NotifyList<Clip> clipList(const TrackId& trackId) const = 0;
 
     virtual void reload() = 0;
 
-    virtual void onTrackAdded(const Track& track) = 0;
-    virtual void onTrackChanged(const Track& track) = 0;
-    virtual void onTrackRemoved(const Track& track) = 0;
+    virtual void notifyAboutTrackAdded(const Track& track) = 0;
+    virtual void notifyAboutTrackChanged(const Track& track) = 0;
+    virtual void notifyAboutTrackRemoved(const Track& track) = 0;
+    virtual void notifyAboutTrackInserted(const Track& track, int pos) = 0;
 
-    virtual void onClipChanged(const Clip& clip) = 0;
-    virtual void onClipAdded(const Clip& clip) = 0;
-    virtual void onClipRemoved(const Clip& clip) = 0;
+    virtual void notifyAboutClipChanged(const Clip& clip) = 0;
+    virtual void notifyAboutClipAdded(const Clip& clip) = 0;
+    virtual void notifyAboutClipRemoved(const Clip& clip) = 0;
 
     virtual TimeSignature timeSignature() const = 0;
     virtual void setTimeSignature(const TimeSignature& timeSignature) = 0;
     virtual muse::async::Channel<TimeSignature> timeSignatureChanged() const = 0;
 
+    virtual muse::async::Channel<> tracksChanged() const = 0;
```

Comment:
oh wait, empty channel? I think we should send tracks as argument or change it to notification

Reply:
I've added tracks to tracksChanged()